implementation/exporters/src/main/java/io/smallrye/opentelemetry/implementation/exporters/sender/VertxHttpSender.java

```diff
@@ -0,0 +1,311 @@
+package io.smallrye.opentelemetry.implementation.exporters.sender;
+
+import static io.smallrye.opentelemetry.implementation.exporters.OtlpExporterUtil.getPort;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.net.URI;
+import java.time.Duration;
+import java.util.Map;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.RejectedExecutionException;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.function.Consumer;
+import java.util.function.Predicate;
+import java.util.function.Supplier;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.zip.GZIPOutputStream;
+
+import io.opentelemetry.exporter.internal.http.HttpSender;
+import io.opentelemetry.exporter.internal.marshal.Marshaler;
+import io.opentelemetry.sdk.common.CompletableResultCode;
+import io.opentelemetry.sdk.internal.ThrottlingLogger;
+import io.smallrye.mutiny.Uni;
+import io.smallrye.opentelemetry.implementation.exporters.BufferOutputStream;
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Handler;
+import io.vertx.core.Vertx;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.http.HttpClient;
+import io.vertx.core.http.HttpClientOptions;
+import io.vertx.core.http.HttpClientRequest;
+import io.vertx.core.http.HttpClientResponse;
+import io.vertx.core.http.HttpMethod;
+import io.vertx.core.tracing.TracingPolicy;
+
+public final class VertxHttpSender implements HttpSender {
+
+    public static final String TRACES_PATH = "/v1/traces";
+    public static final String METRICS_PATH = "/v1/metrics";
+    public static final String LOGS_PATH = "/v1/logs";
+
+    private static final Logger internalLogger = Logger.getLogger(VertxHttpSender.class.getName());
+    private static final ThrottlingLogger logger = new ThrottlingLogger(internalLogger);
+
+    private static final int MAX_ATTEMPTS = 3;
+
+    private final String basePath;
+    private final boolean compressionEnabled;
+    private final Map<String, String> headers;
+    private final String contentType;
+    private final HttpClient client;
+    private final String signalPath;
+
+    public VertxHttpSender(
```

Comment:
Maybe we should publish an artifact with just these senders, so we could use them in the other projects

Reply:
Perhaps. It would be nice if these were actually shared between Quarkus and WF so they're maintained a bit more actively. I understand the wiring's much different with the Quarkus build, but a guy can dream, right? :P That said, something like that sounds like a post MP Tel 2 support kind of change.

Reply:
I can live with that :)
Do you want to create an issue for later?